codegen-renderers/src/main/kotlin/io/vrap/codegen/languages/java/model/second/JavaObjectTypeClassRenderer.kt

```diff
@@ -0,0 +1,150 @@
+                |    }
+                |
+                |    @Override
+                |    public boolean equals(Object o) {
+                |        return EqualsBuilder.reflectionEquals(this, o);
+                |    }
+                |
+                |    @Override
+                |    public int hashCode() {
+                |        return HashCodeBuilder.reflectionHashCode(this);
+                |    }
+                |}
+        """.trimMargin().keepIndentation()
+
+        return TemplateFile(
+            relativePath = "${vrapType.`package`}.${vrapType.simpleClassName}Impl".replace(".", "/") + ".java",
+            content = content
+        )
+    }
+
+    fun Property.toJavaField(): String {
+
+        return if (this.isPatternProperty()) {
+            "private Map<String,${this.type.toVrapType().simpleName()}> values;"
+        } else {
+            "private ${this.type.toVrapType().simpleName()} ${if (this.isPatternProperty()) "values" else this.name};"
+        }
+
+    }
+
+    fun ObjectType.toBeanFields() = this.properties
+        .filter { it.name != this.discriminator }
+        .map { it.toJavaField() }.joinToString(separator = "\n\n")
+
+    fun ObjectType.setters() = this.properties
+        //Filter the discriminators because they don't make much sense the generated bean
+        .filter { it.name != this.discriminator }
+        .map { it.setter() }
+        .joinToString(separator = "\n\n")
+
+
+    fun ObjectType.getters() = this.properties
+        //Filter the discriminators because they don't make much sense the generated bean
+        .filter { it.name != this.discriminator }
+        .map { it.getter() }
+        .joinToString(separator = "\n\n")
+
+
+    fun Property.setter(): String {
+        return if (this.isPatternProperty()) {
+
+            """
+            |@JsonAnySetter
+            |public void setValue(String key, ${this.type.toVrapType().simpleName()} value) {
+            |    if (values == null) {
+            |        values = new HashMap<>();
+            |    }
+            |    values.put(key, value);
+            |}
+            """.trimMargin()
+        } else {
+            """
+            |public void set${this.name.capitalize()}(final ${this.type.toVrapType().simpleName()} ${this.name}){
+            |   this.${this.name} = ${this.name};
+            |}
+            """.trimMargin()
+        }
+    }
+
+    fun Property.getter(): String {
+        return if (this.isPatternProperty()) {
+
+            """
+            |${this.validationAnnotations()}
+            |${this.type.toComment()}
+            |@JsonAnyGetter
+            |public Map<String, ${this.type.toVrapType().simpleName()}> values() {
+            |    return values;
+            |}
+            """.trimMargin()
+        } else {
+            """
+            |${this.type.toComment()}
+            |${this.validationAnnotations()}
+            |@JsonProperty("${this.name}")
```

Comment:
I think that we don't have to repeat these jackson annotations here and I would suggest to remove them.